canvas_fe/src/components/users/profileModal.js

```diff
@@ -0,0 +1,51 @@
+import React, { Component, Fragment } from "react";
+import { Modal, ModalBody, ModalHeader } from 'reactstrap';
+import UserProfile from "./profile";
+import { axiosInstance } from "../../constants";
+import { BASE_API_URL } from "../../constants";
+
+
+
+class UserProfileModal extends Component {
+    state = {
+        user_id: 0,
+        isOpen: false
+    };
+
+    toggle = () => {
```

Comment:
1. don't create toggle functionality in this component
2. since the isOpen state is in the parent, put this functionality there
3. then pass a function when called updates the isOpen state